cmd/frontend/db/saved_searches.go

```diff
@@ -0,0 +1,315 @@
+	}
+
+	return false, nil
+
+}
+
+// ListAll lists all the saved searches on an instance.
+//
+// 🚨 SECURITY: This method does NOT verify the user's identity or that the user is an admin.
+// It is the caller's responsibility to make sure this response never makes it to a user.
+func (s *savedSearches) ListAll(ctx context.Context) (savedSearches []api.SavedQuerySpecAndConfig, err error) {
+	if Mocks.SavedSearches.ListAll != nil {
+		return Mocks.SavedSearches.ListAll(ctx)
+	}
+
+	tr, ctx := trace.New(ctx, "db.SavedSearches.ListAll", "")
+	defer func() {
+		tr.SetError(err)
+		tr.LogFields(otlog.Int("count", len(savedSearches)))
+		tr.Finish()
+	}()
+
+	q := sqlf.Sprintf(`SELECT id,
+		description,
+		query,
+		notify_owner,
+		notify_slack,
+		user_id,
+		org_id,
+		slack_webhook_url FROM saved_searches
+	`)
+	rows, err := dbconn.Global.QueryContext(ctx, q.Query(sqlf.PostgresBindVar))
+	if err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		var sq api.SavedQuerySpecAndConfig
+		if err := rows.Scan(
+			&sq.Config.Key,
+			&sq.Config.Description,
+			&sq.Config.Query,
+			&sq.Config.Notify,
+			&sq.Config.NotifySlack,
+			&sq.Config.UserID,
+			&sq.Config.OrgID,
+			&sq.Config.SlackWebhookURL); err != nil {
+			return nil, err
+		}
+		sq.Spec.Key = sq.Config.Key
+		if sq.Config.UserID != nil {
+			sq.Spec.Subject.User = sq.Config.UserID
+		} else if sq.Config.OrgID != nil {
+			sq.Spec.Subject.Org = sq.Config.OrgID
+		}
+		savedSearches = append(savedSearches, sq)
+	}
+	return savedSearches, nil
+}
+
+func (s *savedSearches) GetByID(ctx context.Context, id int32) (*api.SavedQuerySpecAndConfig, error) {
+	if Mocks.SavedSearches.GetByID != nil {
+		return Mocks.SavedSearches.GetByID(ctx, id)
+	}
+	var savedSearch api.SavedQuerySpecAndConfig
+
+	err := dbconn.Global.QueryRowContext(ctx, `SELECT
+		id,
+		description,
+		query,
+		notify_owner,
+		notify_slack,
+		user_id,
+		org_id,
+		slack_webhook_url
+		FROM saved_searches WHERE id=$1`, id).Scan(
+		&savedSearch.Config.Key,
+		&savedSearch.Config.Description,
+		&savedSearch.Config.Query,
+		&savedSearch.Config.Notify,
+		&savedSearch.Config.NotifySlack,
+		&savedSearch.Config.UserID,
+		&savedSearch.Config.OrgID,
+		&savedSearch.Config.SlackWebhookURL)
+	savedSearch.Spec.Key = savedSearch.Config.Key
+	if savedSearch.Config.UserID != nil {
+		savedSearch.Spec.Subject = api.SettingsSubject{User: savedSearch.Config.UserID}
+	} else if savedSearch.Config.OrgID != nil {
+		savedSearch.Spec.Subject = api.SettingsSubject{Org: savedSearch.Config.OrgID}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return &savedSearch, err
+}
+
+// ListSavedSearchesByUserID lists all the saved searches associated with a user,
+// including saved searches in organizations the user is a member of.
+//
+// 🚨 SECURITY: This method does NOT verify the user's identity or that the user is an admin.
+// It is the caller's responsibility to make sure this method returns only saved searches associated with
```

Comment:
```suggestion
// It is the caller's responsibility to make sure this response never makes it to a user
// without proper permissions.
```

Your current wording makes it sound like this method may return results not for `userID` but for some other arbitrary user, which isn't the case.

Reply:
Ah I see I am wrong and it is partially the case since it is returning all associated saved searches. Nevermind.